### PR TITLE
Fixed the warnings in light example from connectedhomeip.

### PR DESCRIPTION
**Pull request by shripad621git**
- Problem: To fix the warnings that were encountered in the light example from connectedhomeip.

-  For issue[ #163](https://github.com/espressif/esp-matter/issues/163) fixed the warnings from connectedhomeip.

- The build of the light example was successful with no warnings from connectedhomeip.

### CODE REVIEW

**Review by Damian-Nordic — CHANGES_REQUESTED** [review]
Initializing `Optional` and `Nullable` is wrong. The whole point of these classes is to be able to distinguish whether a given value is specified or not, so initializing them to 0 will likely change the behavior of the code. The compilers tend to generate false-positive `maybe-uninitalized` warnings for `Optional` objects because they cannot figure out the relationship between `HasValue()` and `Value()` methods, but if that is a problem I would suggest simply removing this warning, using `-Wno-error=maybe-uninitialized`, or adding some `#pragma` to `Optional.h` to suppress this warning locally.

File: src/app/util/attribute-storage.cpp [hunk 1]
```diff
@@ -114,8 +114,7 @@ void emberAfEndpointConfigure()
 {
     uint16_t ep;
 
-    static_assert(FIXED_ENDPOINT_COUNT <= std::numeric_limits<decltype(ep)>::max(),
-                  "FIXED_ENDPOINT_COUNT must not exceed the size of the endpoint data type");
+    assert(FIXED_ENDPOINT_COUNT <= std::numeric_limits<decltype(ep)>::max());
```
Comment (tcarmelveilleux) [inline src/app/util/attribute-storage.cpp:117]: Cannot change static asserts for runtime asserts

**Review by tcarmelveilleux — CHANGES_REQUESTED** [review]
The fixes are not really fixes, they seem to be a trial to make it build in a configuration where other build configs are improperly set.

File: src/app/EventManagement.cpp [hunk 2]
```diff
@@ -407,7 +407,7 @@ CHIP_ERROR EventManagement::LogEventPrivate(EventLoggingDelegate * apDelegate, c
     CHIP_ERROR err               = CHIP_NO_ERROR;
     uint32_t requestSize         = 0;
     aEventNumber                 = 0;
-    CircularTLVWriter checkpoint = writer;
+    CircularTLVWriter checkpoint;
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/EventManagement.cpp:410]: This is _definitely_ wrong.  The whole point of `checkpoint` is to copy the state of `writer` so we can restore it later if we need to.

File: src/app/clusters/window-covering-server/window-covering-server.cpp [hunk 3]
```diff
@@ -338,7 +338,7 @@ LimitStatus CheckLimitState(uint16_t position, AbsoluteLimits limits)
 
 bool IsPercent100thsValid(Percent100ths percent100ths)
 {
-    if (CHECK_BOUNDS_VALID(WC_PERCENT100THS_MIN_OPEN, percent100ths, WC_PERCENT100THS_MAX_CLOSED))
+    if ((WC_PERCENT100THS_MIN_OPEN< percent100ths)||(percent100ths> WC_PERCENT100THS_MAX_CLOSED))
```
Comment (bzbarsky-apple) [inline src/app/clusters/window-covering-server/window-covering-server.cpp:341]: Why this change?  And after this CHECK_BOUNDS_VALID is unused, so why is it still there if we do make this change?

File: src/app/util/attribute-storage.cpp [hunk 4]
```diff
@@ -139,7 +138,7 @@ void emberAfEndpointConfigure()
 
     emberEndpointCount                = FIXED_ENDPOINT_COUNT;
     DataVersion * currentDataVersions = fixedEndpointDataVersions;
-    for (ep = 0; ep < FIXED_ENDPOINT_COUNT; ep++)
+    for (ep = 0; ep < emberEndpointCount; ep++)
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.cpp:141]: Why this change?  Given line 139, this should be no different, I would think, unless something inside the loop changes `emberEndpointCount`... but if it does, that's because someone is adding dynamic endpoints, and we do _not_ want to do this init for dynamic endpoints: it will cause out-of-bounds array accesses!

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
